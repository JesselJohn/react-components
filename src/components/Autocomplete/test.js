import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

import TestComponent from "./testcomponent";

describe("Root Component Tests", () => {
  beforeAll(() => shallow(React.createElement(TestComponent)));

  it("To Match SnapShot", () => {
    expect(shallowToJson(TestComponent)).toMatchSnapshot();
  });
});
