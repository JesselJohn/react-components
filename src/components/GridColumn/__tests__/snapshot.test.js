import React from "react";
import { shallow, mount } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

import { GridColumnSuccessTest, GridColumnErrorTest } from "../test.component";

describe("Root Component Tests", () => {
  it("Matches Shallow SnapShot", done => {
    expect(shallowToJson(shallow(<GridColumnSuccessTest />))).toMatchSnapshot();
    expect(shallowToJson(shallow(<GridColumnErrorTest />))).toMatchSnapshot();
    done();
  });

  it("Matches Rendered SnapShot", done => {
    const successMounted = mount(<GridColumnSuccessTest />);
    const errorMounted = mount(<GridColumnErrorTest />);

    setTimeout(() => {
      expect(successMounted.find(".nwc-grid-col").length).toBe(1);
      expect(successMounted).toMatchSnapshot();

      expect(errorMounted.find(".nwc-grid-col").length).toBe(2);
      expect(errorMounted).toMatchSnapshot();

      done();
    }, 300);
  });
});
