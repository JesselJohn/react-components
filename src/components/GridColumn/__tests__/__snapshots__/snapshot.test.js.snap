// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root Component Tests Matches Rendered SnapShot 1`] = `
<GridColumnSuccessTest>
  <GridColumn
    className=""
  >
    <div
      className="nwc-grid-col "
    >
      <div>
        Hello
      </div>
    </div>
  </GridColumn>
</GridColumnSuccessTest>
`;

exports[`Root Component Tests Matches Rendered SnapShot 2`] = `
<GridColumnErrorTest>
  <GridColumn
    className=""
  >
    <div
      className="nwc-grid-col "
    >
      <GridColumn
        className=""
      >
        <div
          className="nwc-grid-col "
        >
          Hello
        </div>
      </GridColumn>
    </div>
  </GridColumn>
</GridColumnErrorTest>
`;

exports[`Root Component Tests Matches Shallow SnapShot 1`] = `
<GridColumn
  className=""
>
  <div>
    Hello
  </div>
</GridColumn>
`;

exports[`Root Component Tests Matches Shallow SnapShot 2`] = `
<GridColumn
  className=""
>
  <GridColumn
    className=""
  >
    Hello
  </GridColumn>
</GridColumn>
`;
